doc/sphinx/language/core/assumptions.rst

```diff
@@ -38,12 +38,11 @@ variable can be introduced at the same time. It is also possible to give
 the type of the variable as follows:
 :n:`(@ident : @type := @term)`.
 
-`(x : T | P)` is syntactic sugar for `(x : Stdlib.Init.Specif.sig (fun x : T => P))`,
+`(x : T | P)` is syntactic sugar for `(x : @Stdlib.Init.Specif.sig T (fun x : T => P))`,
```

Comment:
should it be this or
```suggestion
`(x : T | P)` is syntactic sugar for `(x : @Stdlib.Init.Specif.sig _ (fun x : T => P))`,
```
?

Reply:
The current description (with `T` not `_`) is more technically correct (`CAst.make ~loc @@ CAppExpl ((sigref loc,None), [t; mkLambdaC ([id], default_binder_kind, t, c)])`).  And since this will never be used for printing, AFAIU, the only difference would be whether we make type inference do extra work.

Reply:
>The current description (with T not _) is more technically correct 

I meant should we also use a hole in the implementation

Reply:
There could be slightly surprising effects from duplicating the term
For instance any warnings from elaborating it would get doubled

Reply:
> For instance any warnings from elaborating it would get doubled

That's a good remark.

On the other side, could that unshare universes, or add extra evar constraints?

This is probably worth experimenting.

As a comparison, we have `Notation "{ x : A | P }" := (sig (A:=A) (fun x => P))` so we may want in any cases to apply the same scheme in both situations.

Reply:
> On the other side, could that unshare universes, or add extra evar constraints?

Definitely.

> This is probably worth experimenting.

This is a barely used feature, so the precise implementation in unlikely to matter, but I'd argue against duplication in any case. Why not simply keep the old behaviour?